kernel/src/process_loading.rs

```diff
@@ -114,6 +114,135 @@ impl fmt::Debug for ProcessLoadError {
 // SYNCHRONOUS PROCESS LOADING
 ////////////////////////////////////////////////////////////////////////////////
 
+// load_processes_return_memory() and load_processes_from_flash_return are copies
+// of load_processes_return_memory() and load_processes_from_flash(). These functions
+// are created so that they return the remaining memory to main.rs after loading
+// all the apps initially. This remaining_memory value is then used by the
+// dynamic process loader to find the right spot to load the new app because otherwise
+// we are not privy to this information.
+
+#[inline(always)]
```

Comment:
Why?

Reply:
The current return type for `load_processes` is `Result<(), ProcessLoadError>`. We want the function to return the remaining memory when the board initializes the processes in `main.rs`. Currently, changing the return type breaks the build for all other boards.

What we require:
`.unwrap_or_else(|(err, remaining_memory)| {
        debug!("Error loading processes!");
        debug!("{:?}", err);
        remaining_memory
    });`

What we currently have:
`.unwrap_or_else(|err| {
        debug!("Error loading processes!");
        debug!("{:?}", err);
    });`
   
I have duplicated the functions needed to return the memory for this implementation, the other fix is to change the return type for all the supported boards.

Reply:
I meant why `#[inline(always)]`

But in terms of duplicating a function, it would probably make more sense to just change the boards to ignore the return type instead of duplicating a function just to return something new

Reply:
I removed the inline, forgot to remove it after copying the functions. 

This is just to test if the dynamic process loading works, so I think it is less work to duplicate the functions to prove the idea than to change the return types for all boards.

Reply:
It's a lot more work to maintain duplicate functions

Reply:
Right, but the synchronous functions will go away when we move to async loading. That will remove the need to change the return types. If we make the change now, we will have to revert later which is pointless. The current implementation takes a step forward in enabling dynamic process loading, but is not the end goal.